internal/template-validator/validation/eval_test.go

```diff
@@ -226,18 +222,18 @@ var _ = Describe("Eval", func() {
 			rules := []Rule{
 				{
 					Name:        "disk bus",
-					Rule:        "enum",
-					Path:        "jsonpath::.spec.domain.devices.disks[*].disk.bus",
+					Rule:        EnumRule,
+					Path:        *path.NewOrPanic("jsonpath::.spec.domain.devices.disks[*].disk.bus"),
 					Message:     "testing",
-					Values:      []string{"sata"},
+					Values:      []path.StringOrPath{{Str: "sata"}},
 					JustWarning: true,
 				}, {
 					Name: "rule-2",
-					Rule: "integer",
-					Min:  6,
-					Max:  8,
+					Rule: IntegerRule,
+					Min:  &path.IntOrPath{Int: 6},
+					Max:  &path.IntOrPath{Int: 8},
 					// any legal path is fine
-					Path:    "jsonpath::.spec.domain.cpu.cores",
+					Path:    *path.NewOrPanic("jsonpath::.spec.domain.cpu.cores"),
```

Comment:
Just asking, when the Path is not correct, should't `res.Status[x].Error` be not nil?

Reply:
Yes, but in this test case the Path is correct. It parses correctly, and points to an existing field in the VM structure.
The `vmCirros` VM used in this test has `.spec.domain.cpu = nil`, but that is not an error, because the validator uses a "reference" VM, that contains default values for nil fields. The `cores` value is taken from the reference VM, and the rule is evaluated.

Reply:
ah yes, sorry. I missed there are no " around `*path.NewOrPanic("jsonpath::.spec.domain.cpu.cores")` :)